src/lib/sidenav/sidenav.ts

```diff
@@ -103,6 +103,12 @@ export class MdSidenav implements AfterContentInit {
   /** Mode of the sidenav; whether 'over' or 'side'. */
   @Input() mode: 'over' | 'push' | 'side' = 'over';
 
+  /** Whether the sidenav can be closed with the escape key or not. */
+  @Input()
+  get disableClose(): boolean { return this._disableClose; }
+  set disableClose(value: boolean) { this._disableClose = coerceBooleanProperty(value); }
+  private _disableClose: boolean = false;
```

Comment:
`disableClose` should also prevent close from backdrop clicks.

While you're doing this, can you also change the `@Output` to `backdropClick`? Looks like we missed it when getting rid of dash-case stuff.